scipy/special/_dispatch.py

```diff
@@ -0,0 +1,21 @@
+from scipy._lib._array_api import array_namespace
+from . import _ufuncs
+from ._ufuncs import *
+
+__all__ = _ufuncs.__all__
+
+# functools.wraps doesn't work because:
+# 'numpy.ufunc' object has no attribute '__module__'
+def dispatcher(f_name, n_args):
+    def wrapped(*args, **kwargs):
+        xp = array_namespace(*args[:n_args])
+        if xp.__name__.endswith('numpy'):
```

Comment:
Put all this in `try`...`except AttributeError` in case library doesn't have the special function?

Reply:
Let's avoid that for now I'd suggest, since we have no idea whether other libraries have a `special` submodule that may be incompatible with `scipy.special` signatures or contain something completely different.

Reply:
So just error if it's not one of these libraries, or convert to NumPy, evaluate, and convert back?

Reply:
I'd say convert to numpy and back for now. Once we have a `special` extension in the array API standard (even in a draft version of the standard) I think we can try the `try`-`except`, but for now I think we can only call out to other libraries in cases where we know it'll work.

Reply:
Ok, done.